.css inheritance in themes (#8424)
gitextensions/gitextensions: GitUI/Themes/README.md

> ﻿GitExtensions stores themes in *.css files.

Reviewer: The readme is a great idea, but it needs some work.

It would be great to add sections, may be something like the following:

```md
# Overview

# Themes locations

### Installed app

* Official themes

...

* Custom themes

...

### Portable app

# How to create custom theme
```

It needs to discuss that there are two distinct modes - installed and portable, and each mode handles themes differently. I.e. the portable mode supports only local themes (those placed in .\Themes) folder, whereas the installed app supports official themes and user-defined themes. The official themes are located in the app installation folder (typically `%ProgramFiles(x86)%\GitExtensions\Themes`), and these mustn't be changed.

Author: ok

﻿# Overview

> GitExtensions supports color customization, in particular user can choose between
> default (operating-system-defined), bright or dark theme.

Reviewer: ```suggestion
Git Extensions supports color customization, specifically a user can choose between
a default (operating system defined), a bright or a dark themes.
```

There is a [Wiki page on Dark Theme](https://github.com/gitextensions/gitextensions/wiki/Dark-Theme)
with some background information and a list of known issues.

# Themes location

> Themes are stored in *.css files. Where GitExtensions looks for theme files depends on whether it
> is Installed or Portable version. Both come with a predefined set of themes in

Reviewer: ```suggestion
Themes are stored in *.css files. Where Git Extensions looks for theme files depends on whether it
is an 'Installed' or a 'Portable' version. Both come with a predefined set of themes in
```

`(App install)\Themes` directory.

> In addition the Installed version looks for user-defined themes in
> `%AppData%\Roaming\GitExtensions\GitExtensions\Themes`

Reviewer: ```suggestion
In addition the Installed version looks for user-defined themes in
`%AppData%\Roaming\GitExtensions\GitExtensions\Themes` directory.
```

> For each found theme file ColorSettings page will show an entry in Theme drop down list.

Reviewer: ```suggestion
For each found theme file `ColorSettings` page will show an entry in the Theme dropdown list.
```

## Installed GitExtension application

### Official themes

> Installed version of GitExtensions stores themes in `(App install)\Themes` directory.

Reviewer: ```suggestion
The Installed version of Git Extensions stores themes in `(App install)\Themes` directory.
```

### Custom themes

> Any modifications for personal use should be saved to
> `%AppData%\Roaming\GitExtensions\GitExtensions\Themes`, because any changes to
> `(App install)\Themes` directory will be lost as soon as user updates or reinstalls GitExtensions.

Reviewer: ```suggestion
Any modifications for personal use should be saved to
`%AppData%\Roaming\GitExtensions\GitExtensions\Themes` directory, because any changes to
`(App install)\Themes` directory will be lost as soon as user updates or re-installs the app.
```

## Portable application

> The portable version of GitExtensions does not have a separate directory for user-defined themes.
> It only uses preinstalled themes from `(App install)\Themes` directory.

Reviewer: ```suggestion
The Portable version of Git Extensions does not have a separate directory for user-defined themes.
It only uses pre-installed themes from `(App install)\Themes` directory.
```

# How to create a custom theme

Copy `Themes\dark.css` or `Theme\bright.css` and change color values. Use a text editor, preferably
with .css syntax highlighting and inline color display.

> There is a plugin for VisualStudio to show .css colors inline.

Reviewer: ```suggestion
There is a plugin for Visual Studio to show .css colors inline.
```
